packages/sdk/src/actions/prepareCreateExternalPosition.test.ts

```diff
@@ -0,0 +1,123 @@
+import { test, expect } from "vitest";
+
+import { ALICE, BOB, EXTERNAL_POSITION_MANAGER, KILN_STAKING_CONTRACT, WETH } from "../../tests/constants.js";
+import { toWei } from "../utils/conversion.js";
+import { ExternalPosition } from "../enums.js";
+import { testActions, sendTestTransaction } from "../../tests/globals.js";
+import { prepareCreateExternalPosition } from "./prepareCreateExternalPosition.js";
+import { decodeEventLog, getEventSelector } from "viem";
+import { IKilnStakingPositionLib } from "../../../abis/src/abis/IKilnStakingPositionLib.js";
+
+test("prepare create external position should work correctly", async () => {
+  const vaultOwner = ALICE;
+  const sharesBuyer = BOB;
+
+  const { comptrollerProxy } = await testActions.createTestVault({
+    vaultOwner,
+    denominationAsset: WETH,
+  });
+
+  const depositAmount = toWei(250);
+
+  await testActions.buyShares({
+    comptrollerProxy,
+    sharesBuyer,
+    investmentAmount: depositAmount,
+  });
+
+  const kilnTypeId = 8n;
+
+  const validatorAmount = 1n;
+
+  const { receipt } = await sendTestTransaction({
+    ...prepareCreateExternalPosition({
+      externalPositionManager: EXTERNAL_POSITION_MANAGER,
+      typeId: kilnTypeId,
+      trade: {
+        type: ExternalPosition.KilnStake,
+        callArgs: {
+          validatorAmount,
+          stakingContract: KILN_STAKING_CONTRACT,
+        },
+      },
+    }),
+    account: vaultOwner,
+    address: comptrollerProxy,
+  });
+
+  const externalPositionDeployedForFundEventSelector = getEventSelector(
+    "ExternalPositionDeployedForFund(address indexed comptrollerProxy, address indexed vaultProxy, address externalPosition, uint256 indexed externalPositionTypeId, bytes data)",
+  );
+
+  const externalPositionDeployedForFundEvent = receipt.logs.find(
+    (log) => log.topics[0] === externalPositionDeployedForFundEventSelector,
+  );
+
+  const validatorsAddedEventSelector = getEventSelector(
+    "ValidatorsAdded(address stakingContractAddress, uint256 validatorAmount)",
+  );
+
+  const validatorsAddedEvent = receipt.logs.find((log) => log.topics[0] === validatorsAddedEventSelector);
+
+  const validatorsAddedEventLog = decodeEventLog({
+    abi: IKilnStakingPositionLib,
+    data: validatorsAddedEvent?.data,
+    topics: validatorsAddedEvent?.topics ?? [],
+  });
```

Comment:
There is a `getLogs` function for that in `viem`

Reply:
done